src/main/asciidoc/writeBehind.adoc

```diff
@@ -0,0 +1,127 @@
+
+= Write-Behind Caching
+
+There is always a need to improve the performance of application and sometimes part of this task really requires a little
+effort as compared to results achieved.
+After profiling any application, one can easily single out database as the bottleneck in achieving the next milestone of performance improvement.
+Thus it becomes obvious to use a cache to give a push to application’s performance.
+
+Predominantly there are two cache usage patterns – cache aside & cache through.
+While cache aside pattern leaves the responsibility of synchronizing SOR(System-Of-Record) which is mainly a database
+and the cache to application code, in cache through pattern application code just interacts with the cache level apis.
+It is the cache that interacts with database for reads/writes.
+The cache through pattern on one hand increases the performance,
+on the  other it reduces the code to maintain for database interaction.
+Thus it becomes logical to use a cache in cache-through pattern,
+but still direct calls from users involve the slowest participant i.e. the database.
+
+The foremost reason for using a cache is to gain performance and scale,
+but in cache through pattern the writes still suffers from the latency induced by the database.
+To overcome this limit of database slowness, you can use a cache as write-behind cache.
+This pattern allows application to let the cache asynchronously update the database,
+while application still gets the performance of cache. Rather than writing to the database in the same thread of execution,
+write-behind queues the data to be written at a later time.
+The cache effectively assumes the role of a database which is fast, while database acts as a backup for the cache.
```

Comment:
ugh, we should stay away from such statements "the cache effectively assumes the role of a database". Can't you convey the benefits differently?